WarehouseManagement.Core/Services/ZoneService.cs

```diff
@@ -0,0 +1,234 @@
+﻿using Microsoft.EntityFrameworkCore;
+using WarehouseManagement.Core.Contracts;
+using WarehouseManagement.Core.DTOs.Zone;
+using WarehouseManagement.Infrastructure.Data.Common;
+using WarehouseManagement.Infrastructure.Data.Models;
+
+namespace WarehouseManagement.Core.Services;
+
+public class ZoneService : IZoneService
+{
+    private readonly IRepository repository;
+
+    public ZoneService(IRepository repository)
+    {
+        this.repository = repository;
+    }
+
+    public async Task CreateAsync(ZoneFormDto model, string userId)
+    {
+        if (await this.ExistsByNameAsync(model.Name))
+        {
+            throw new ArgumentException($"Zone with name '{model.Name}' already exists.");
+        }
+
+        Zone zone = new Zone()
```

Comment:
you can also just use `var` instead of the `type` when creating variable. The others are already doing that and i also dont see a reaso to not use `var.